lib/src/equality.dart

```diff
@@ -81,7 +81,7 @@ class EqualityBy<E, F> implements Equality<E> {
 /// Note that [equals] and [hash] take `Object`s rather than `E`s. This allows
 /// `E` to be inferred as `Null` in const contexts where `E` wouldn't be a
 /// compile-time constant, while still allowing the class to be used at runtime.
-class DefaultEquality<E> implements Equality<E> {
+final class DefaultEquality<E> implements Equality<E> {
```

Comment:
https://github.com/angulardart/angular/blob/e943f6f0b8f8ae0d64c2c7935d4554f41d86e2cd/angular/lib/src/runtime/check_binding.dart#L186-L190


This has a usage of `extends`